File: patch-profile.js
import fs from 'fs';

const path = 'src/pages/Profile.tsx';
let code = fs.readFileSync(path, 'utf8');

// 1. Add import
if (!code.includes('deleteAccount')) {
  code = code.replace(
    "import { fetchMyBookings, fetchMyTours, uploadAvatar, initiatePasswordChange, User } from '../api';",
    "import { fetchMyBookings, fetchMyTours, uploadAvatar, initiatePasswordChange, deleteAccount, User } from '../api';"
  );
}

// 2. Add state and handler
const targetState = 'const [targetEmail, setTargetEmail] = useState<string | null>(null);';
const newState = `
  const [isDeletingAccount, setIsDeletingAccount] = useState(false);
  const [showDeleteConfirm, setShowDeleteConfirm] = useState(false);
  const [deleteToursChecked, setDeleteToursChecked] = useState(false);

  const handleDeleteAccount = async () => {
    setIsDeletingAccount(true);
    try {
      await deleteAccount();
      onLogout();
      onNavigate('home');
    } catch (err: any) {
      alert(isKa ? \`შეცდომა წაშლისას: \${err.message}\` : \`Error deleting account: \${err.message}\`);
    } finally {
      setIsDeletingAccount(false);
    }
  };
`;
if (!code.includes('handleDeleteAccount')) {
  code = code.replace(targetState, targetState + "\n" + newState);
}

// 3. Replace danger zone block
const dangerZoneRegex = /<div className="mt-8 p-6 rounded-3xl bg-red-50 border border-red-100">([\s\S]*?)<\/button>\s*<\/div>/;
const newDangerZone = `<div className="mt-8 p-6 rounded-3xl bg-red-50 border border-red-100">
                <h3 className="text-red-800 font-black mb-2 flex items-center gap-2">
                  <span className="material-symbols-outlined">dangerous</span>
                  {isKa ? 'საშიში ზონა' : 'Danger Zone'}
                </h3>
                <p className="text-sm text-red-600 mb-4 font-medium">
                  {isKa ? 'ანგარიშის წაშლა სამუდამოა და ვერ აღდგება.' : 'Deleting your account is permanent and cannot be undone.'}
                </p>

                {showDeleteConfirm ? (
                  <div className="bg-white p-4 rounded-xl border border-red-200 mt-4 animate-fade-in shadow-sm">
                    <p className="text-red-700 font-bold mb-3 text-sm flex items-center gap-2">
                      <span className="material-symbols-outlined text-[18px]">warning</span>
                      {isKa ? 'დარწმუნებული ხართ?' : 'Are you completely sure?'}
                    </p>
                    
                    {user.role === 'operator' && myTours.length > 0 && (
                      <label className="flex items-start gap-3 p-3 bg-red-50 rounded-lg mb-4 cursor-pointer hover:bg-red-100/50 transition-colors border border-red-100">
                        <input 
                          type="checkbox" 
                          checked={deleteToursChecked}
                          onChange={(e) => setDeleteToursChecked(e.target.checked)}
                          className="mt-1 w-4 h-4 text-red-600 rounded border-red-300 focus:ring-red-500 cursor-pointer"
                        />
                        <span className="text-xs text-red-800 font-semibold leading-relaxed">
                          {isKa 
                            ? \`ვადასტურებ, რომ ჩემი \${myTours.length} ტური სამუდამოდ წაიშლება ანგარიშთან ერთად.\` 
                            : \`I acknowledge that my \${myTours.length} published tours will be permanently deleted along with my account.\`}
                        </span>
                      </label>
                    )}

                    <div className="flex gap-3">
                      <button 
                        onClick={() => { setShowDeleteConfirm(false); setDeleteToursChecked(false); }}
                        className="flex-1 py-2.5 px-4 bg-gray-100 hover:bg-gray-200 text-gray-700 font-bold rounded-lg text-xs transition-colors"
                        disabled={isDeletingAccount}
                      >
                        {isKa ? 'გაუქმება' : 'Cancel'}
                      </button>
                      <button 
                        onClick={handleDeleteAccount}
                        disabled={isDeletingAccount || (user.role === 'operator' && myTours.length > 0 && !deleteToursChecked)}
                        className="flex-1 py-2.5 px-4 bg-red-600 hover:bg-red-700 disabled:bg-red-300 disabled:cursor-not-allowed text-white font-black rounded-lg text-xs shadow-md shadow-red-500/20 transition-all flex items-center justify-center gap-2"
                      >
                        {isDeletingAccount ? (
                          <span className="w-4 h-4 border-2 border-white border-t-transparent rounded-full animate-spin"></span>
                        ) : (
                          <>
                            <span className="material-symbols-outlined text-[14px]">delete_forever</span>
                            {isKa ? 'სამუდამოდ წაშლა' : 'Delete Permanently'}
                          </>
                        )}
                      </button>
                    </div>
                  </div>
                ) : (
                  <button 
                    onClick={() => setShowDeleteConfirm(true)}
                    className="text-red-700 font-black text-sm flex items-center gap-2 hover:underline bg-white px-4 py-2 rounded-lg border border-red-200 shadow-sm transition-all hover:shadow-md w-fit"
                  >
                    <span className="material-symbols-outlined text-[18px]">delete</span>
                    {isKa ? 'ანგარიშის წაშლა' : 'Delete Account'}
                  </button>
                )}
              </div>`;

code = code.replace(dangerZoneRegex, newDangerZone);

fs.writeFileSync(path, code);
console.log('Profile UI patched.');

File: patch-profile.py
import os

file_path = 'src/pages/Profile.tsx'
with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# 1. Replace handleDeleteTour
old_delete = """  const handleDeleteTour = async (id: number) => {
    if (!window.confirm(isKa ? 'ნამდვილად გსურთ წაშლა?' : 'Are you sure you want to delete this tour?')) return;
    try {
      await deleteTour(id);
      setMyTours(myTours.filter(t => t.id !== id));
      alert(isKa ? 'წარმატებით წაიშალა' : 'Tour deleted successfully');
    } catch (err) {
      alert(isKa ? 'შეცდომა წაშლისას' : 'Failed to delete tour');
    }
  };"""

new_delete = """  const promptDeleteTour = (id: number) => {
    setTourToDelete(id);
  };

  const confirmDeleteTour = async () => {
    if (tourToDelete === null) return;
    setIsDeletingTour(true);
    try {
      await deleteTour(tourToDelete);
      setMyTours(myTours.filter(t => t.id !== tourToDelete));
      setToast({ message: isKa ? 'წარმატებით წაიშალა' : 'Tour deleted successfully', type: 'success' });
    } catch (err) {
      setToast({ message: isKa ? 'შეცდომა წაშლისას' : 'Failed to delete tour', type: 'error' });
    } finally {
      setIsDeletingTour(false);
      setTourToDelete(null);
    }
  };"""

# 2. Replace handleUpdateProfile
old_update = """  const handleUpdateProfile = async (e: React.FormEvent) => {
    e.preventDefault();
    setIsSaving(true);
    try {
      await updateMe(editForm);
      onUpdateUser(editForm);
      alert(isKa ? 'პროფილი წარმატებით განახლდა' : 'Profile updated successfully');
    } catch (err) {
      alert(isKa ? 'შეცდომა განახლებისას' : 'Failed to update profile');
    } finally {
      setIsSaving(false);
    }
  };"""

new_update = """  const handleUpdateProfile = async (e: React.FormEvent) => {
    e.preventDefault();
    setIsSaving(true);
    try {
      await updateMe(editForm);
      onUpdateUser(editForm);
      setToast({ message: isKa ? 'პროფილი წარმატებით განახლდა' : 'Profile updated successfully', type: 'success' });
    } catch (err) {
      setToast({ message: isKa ? 'შეცდომა განახლებისას' : 'Failed to update profile', type: 'error' });
    } finally {
      setIsSaving(false);
    }
  };"""

# 3. Handle AnimatePresence
old_modal = """      <AnimatePresence>
        {toast && ("""

new_modal = """      <AnimatePresence>
        {tourToDelete !== null && (
          <motion.div
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
            className="fixed inset-0 z-[100] flex items-center justify-center p-4 bg-black/40 backdrop-blur-sm"
          >
            <motion.div
              initial={{ scale: 0.95, opacity: 0 }}
              animate={{ scale: 1, opacity: 1 }}
              exit={{ scale: 0.95, opacity: 0 }}
              className="bg-white rounded-3xl p-6 sm:p-8 max-w-sm w-full shadow-2xl border border-border-light relative overflow-hidden"
            >
              <div className="absolute top-0 left-0 w-full h-2 bg-red-500"></div>
              <div className="w-16 h-16 bg-red-50 rounded-full flex items-center justify-center mx-auto mb-4">
                <span className="material-symbols-outlined text-red-500 text-3xl">delete_sweep</span>
              </div>
              <h3 className="text-xl font-black text-center text-text-main mb-2">
                {isKa ? 'გსურთ ტურის წაშლა?' : 'Delete this tour?'}
              </h3>
              <p className="text-sm text-text-muted text-center font-medium mb-6">
                {isKa ? 'ეს ქმედება სამუდამოა და ვეღარ აღადგენთ.' : 'This action is permanent and cannot be undone.'}
              </p>
              <div className="flex gap-3">
                <button
                  onClick={() => setTourToDelete(null)}
                  disabled={isDeletingTour}
                  className="flex-1 py-3 px-4 bg-gray-100 hover:bg-gray-200 text-gray-700 font-bold rounded-xl transition-colors"
                >
                  {isKa ? 'გაუქმება' : 'Cancel'}
                </button>
                <button
                  onClick={confirmDeleteTour}
                  disabled={isDeletingTour}
                  className="flex-1 py-3 px-4 bg-red-500 hover:bg-red-600 disabled:bg-red-300 text-white font-black rounded-xl shadow-md shadow-red-500/20 transition-all flex items-center justify-center gap-2"
                >
                  {isDeletingTour ? (
                    <span className="w-5 h-5 border-2 border-white border-t-transparent rounded-full animate-spin"></span>
                  ) : (
                    <>
                      <span className="material-symbols-outlined text-[18px]">delete</span>
                      {isKa ? 'წაშლა' : 'Delete'}
                    </>
                  )}
                </button>
              </div>
            </motion.div>
          </motion.div>
        )}
      </AnimatePresence>

      <AnimatePresence>
        {toast && ("""

# Replace preserving windows line endings by fixing string matching
old_delete = old_delete.replace('\\n', '\\r\\n')
new_delete = new_delete.replace('\\n', '\\r\\n')
old_update = old_update.replace('\\n', '\\r\\n')
new_update = new_update.replace('\\n', '\\r\\n')
old_modal = old_modal.replace('\\n', '\\r\\n')
new_modal = new_modal.replace('\\n', '\\r\\n')

if old_delete in content:
    content = content.replace(old_delete, new_delete)
    print("Replaced deleteTour")
else:
    print("Could not find deleteTour using windows CRLF, trying unix LF")
    content = content.replace(old_delete.replace('\\r\\n', '\\n'), new_delete.replace('\\r\\n', '\\n'))

if old_update in content:
    content = content.replace(old_update, new_update)
    print("Replaced updateProfile")
else:
    content = content.replace(old_update.replace('\\r\\n', '\\n'), new_update.replace('\\r\\n', '\\n'))
    
if old_modal in content:
    content = content.replace(old_modal, new_modal)
    print("Replaced modal")
else:
    content = content.replace(old_modal.replace('\\r\\n', '\\n'), new_modal.replace('\\r\\n', '\\n'))

content = content.replace("onClick={() => handleDeleteTour(tour.id)}", "onClick={() => promptDeleteTour(tour.id)}")

with open(file_path, 'w', encoding='utf-8', newline='') as f:
    f.write(content)

# Use regex to find out what might be breaking.
print("Done patching.")
